astrosat/migrations/0005_databaselogrecord_uuid.py

```diff
@@ -0,0 +1,19 @@
+# Generated by Django 3.1.3 on 2020-11-11 14:44
+
+from django.db import migrations, models
+import uuid
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('astrosat', '0004_astrosatsettings_enable_debug_toolbar'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='databaselogrecord',
+            name='uuid',
+            field=models.UUIDField(default=uuid.uuid4, editable=False),
+        ),
+    ]
```

Comment:
Do you think this new `uuid` field should be unique?  I don't know.  I suppose it usually will.  But I don't enforce it (via a `unique=True` argument).  This is a bit unusual for `UUIDField`, but not necessarily a bad thing.  

Because of the mechanics of Django it means that:
* if we keep it not unique then the first time this migration gets run all existing records will have the same uuid
* if we change it to be unique then there will have to be multiple migrations to cope w/ that

Reply:
I guess i'd defer to your experience, but I would normally say having all existing records have the same uuid is bad, but in practice, we don't have many records, so it isn't a big deal. Also, the uuid is only really relevant within the boundaries of a single POST request/response. We don't need it for anything else, so I don't think it is a big deal.

Reply:
I'm going to say that we _should_ ensure that every record has a unique uuid.  

In order to do this you want to use 3 migrations:  one to add the field without making it unique yet, one to give a unique value to each existing instance, and one to apply the unique constraint.  Therefore you will have to do the following:

1. roll back the current migration: `manage.py migrate astrosat 0004`
2. manually modify "astrosat/migrations/0005_databaselogrecord_uuid.py" so that the fields are not unique but are nullable
3. create a new blank data migration: `manage.py makemigrations --empty --name databaselogrecord_uuid_populate astrosat` (I like using the "name" argument to indicate what the migration is for)
4. manaully update that newly created data migration to call the following function as an operation:
```
def populate_uuids(apps, schema_editor):
    DatabaseLogRecordModel = apps.get_model('astrosat', 'DatabaseLogRecord')
    for model in DatabaseLogRecordModel.objects.all():
        model.uuid = uuid.uuid4()
        model.save(update_fields=["uuid"])
```
5. create another blank migration: `manage.py makemigrations --empty --name databaselogrecord_uuid_unique`
6. copy the `operations` section of your original (0005) migration but this time you want to call `AlterField` instead of `AddField` and ensure that `unique=True`
7. finally run all these migrations: `manage.py migrate`

Reply:
Incidentally, you can see an example of using 3 migrations to accomplish this at: https://github.com/astrosat/covid-app/pull/222